### PR TITLE
Added option to make product id a selectable field

### PR DESCRIPTION
**Pull request by joram77**
### Ticket
- [shop-analytics: make product id a selectable field](https://app.asana.com/0/809933051638353/1200535201752926/f)

### CODE REVIEW

File: templates/settings.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 global $wp_roles;
 
 $product_attributes = function_exists('wc_get_attribute_taxonomies') ? wc_get_attribute_taxonomies() : '';
+$shop_analytics_product_id = get_option('shop_analytics_product_id', 'post_id');
```
Comment (fabianmarz) [inline templates/settings.php:8]: I think it's fine to remove the `shop_analytics` prefix for the variable name as we are already in the `ShopAnalytics` namespace. The option name should remain the same however

```suggestion
$product_id = get_option('shop_analytics_product_id', 'post_id');
```

File: src/WooCommerce.php [hunk 2]
```diff
@@ -74,9 +74,16 @@ public static function getProductDetails($product_id = 0, $primary_category = TR
       $product_name = str_replace(["'", '"'], '', wp_strip_all_tags($product->get_name(), TRUE));
     }
 
+    $product_id_field = get_option('shop_analytics_product_id');
+
```
Comment (fabianmarz) [inline src/WooCommerce.php:78]:
```suggestion
```

**Review by fabianmarz — COMMENTED** [review]
Looks good so far. Please wait until @daschmidt94 has tested it as I don't know about it fits the tracking needs.

**Review by Mauricio-Urrego — APPROVED** [review]
Looks like this passed functional QA now which was the only request by @fabianmarz. Code still looks good.

**Pull request merged**